### DISCUSSION TITLE
Localized Navbar Links

### DISCUSSION THREAD

**Discussion by anaclumos** [facebook/docusaurus#9218] [General]
Is it possible to have different links on navbar for locales?

**Comment by slorber** [comment]
It's probably not ideal but you can use the `process.env.DOCUSAURUS_CURRENT_LOCALE` env variable for that:

https://github.com/facebook/docusaurus/issues/4542#issuecomment-1434839071

**Reply by anaclumos** [reply]
Within `docusaurus.config.js`

```ts
console.log('CURRENT_LOCALE', process.env.DOCUSAURUS_CURRENT_LOCALE)
```

gives

```
CURRENT_LOCALE undefined
```

Is there any further configuration needed?

**Reply by slorber** [reply]
The GitHub comment I linked to explains how it works and why you get an undefined value there

**Reply by anaclumos** [reply]
Yes, `You run docusaurus build, only after the first load, once we start iterating over locales
` is slightly confusing for me. Would you mind to explain?

**Reply by slorber** [reply]
The JS file has to be loaded first to know the list of locales your site support, and during that process the env variable will be undefined. Somehow we load the config file twice: once with undefined, and once with the current locale value.

**Reply by anaclumos** [reply]
Interesting... because I'm doing like this

https://github.com/anaclumos/extracranial/blob/9194b501de88b3a3c1d41bc76c553f6488c6c7ff/config/navbar.config.ts#L16

And Docusaurus doesn't seem to recognize `process.env.DOCUSAURUS_CURRENT_LOCALE`. When I build it like this, it will fallback to English (default). Am I missing something?

**Reply by slorber** [reply]
Try to build, it should work normally.

If you log the value, it will be logged twice. The first one will be undefined but this undefined value will not be used so it's not important

**Reply by anaclumos** [reply]
The `process.env.DOCUSAURUS_CURRENT_LOCALE` didn't work for me, but I also found a hacky way.

Put two links side by side and hide other locales with `:lang()` CSS property.

```
.navbar-en-link:lang(ko-KR) {
  display: none;
}
```

**Reply by slorber** [reply]
CSS can work yes

Show me a small repro of using `DOCUSAURUS_CURRENT_LOCALE` and I'll fix your code. If I don't see your code (I mean, not a huge complex existing repo) it's hard to see where you fail.
